client/src/main/scala/drt/client/components/EditKeyCloakUser.scala

```diff
@@ -0,0 +1,115 @@
+package drt.client.components
+
+import java.util.UUID
+
+import drt.client.actions.Actions.SaveUserGroups
+import drt.client.services._
+import drt.shared.AirportConfigs
+import drt.shared.KeyCloakApi.KeyCloakUser
+import japgolly.scalajs.react.extra.Reusability
+import japgolly.scalajs.react.vdom.html_<^._
+import japgolly.scalajs.react.{Callback, ReactEventFromInput, ScalaComponent}
+
+
+object EditKeyCloakUser {
+
+  case class Props(user: KeyCloakUser, groups: Set[String])
+
+  case class State(groups: Set[String])
+
+  implicit val keycloakUser: Reusability[KeyCloakUser] = Reusability.derive[KeyCloakUser]
+  implicit val stateReuse: Reusability[State] = Reusability.derive[State]
+  implicit val propsReuse: Reusability[Props] = Reusability.derive[Props]
+  val component = ScalaComponent.builder[Props]("EditKeyCloakUser")
+    .initialStateFromProps(p => State(p.groups))
+    .renderPS((scope, props, state) => {
+
+      val ports = AirportConfigs.portGroups
+
+      val otherGroups = List(
+        "Staff Admin"
+      )
+
+      def groupSelector(selectionGroups: List[String], selected: Set[String]) = selectionGroups.map(g =>
+        <.div(^.className :="checkbox", <.label(<.input.checkbox(
+          ^.value := g,
+          ^.checked := selected.contains(g),
+          ^.onChange ==> ((e: ReactEventFromInput) => {
+            val target = e.target
+            scope.modState(s => {
+              if (target.checked)
+                s.copy(groups = s.groups + g)
+              else
+                s.copy(groups = s.groups.filter(_ != g))
```

Comment:
Would a filterNot be more natural here? I'm never quite sure but it might read easier